checks.c

```diff
@@ -501,13 +505,16 @@ static void CheckNameEntryValid(X509_NAME_ENTRY *ne)
 		}
 	}
 
-	if (nid == NID_pkcs9_emailAddress)
+	if (nid == NID_pkcs9_emailAddress || nid == NID_domainComponent)
 	{
 		if (data->type != V_ASN1_IA5STRING)
 		{
 			SetError(ERR_INVALID_NAME_ENTRY_TYPE);
 		}
 	}
+	else if (nid == NID_pkcs9_unstructuredName && data->type == V_ASN1_IA5STRING) {
```

Comment:
I like the { on the next line

Reply:
I think you want:
```
else if (nid == NID_pkcs9_unstructuredName)
{
   if (data->type == V_ASN1_IA5STRING &&
      (data->type != V_ASN1_PRINTABLESTRING) &&
      (data->type != V_ASN1_UTF8STRING) &&
      (data->type != V_ASN1_T61STRING) &&
      (data->type != V_ASN1_UNIVERSALSTRING) &&
      (data->type != V_ASN1_BMPSTRING))
[...]
```

Reply:
I guess your way works too and is much simpler.

Reply:
{ moved to next line by https://github.com/kroeckx/x509lint/pull/25/commits/c834d9ddbe59f6a7d0791bc816239408923e4dac

Reply:
I wanted to avoid duplicating the DirectoryString type checks.  I suppose we could fold them into a function though.  Would you prefer that?